src/relay.cpp

```diff
@@ -591,6 +692,13 @@ void relay_relay_forw(int sock, const uint8_t *msg, int32_t len, const ip6_hdr *
     target_addr.sin6_flowinfo = 0;
     target_addr.sin6_port = htons(CLIENT_PORT);
     target_addr.sin6_scope_id = if_nametoindex(config->interface.c_str());
+    int sock = config->lla_sock;
+    if (isIPv6Zero(&dhcp_relay_header->link_address)) {
+        // In this case, it's multi-level relay
```

Comment:
Why can't we use the Interface-Id option for both the dual tor and non-dual tor scenarios? This approach seems clear to me.  Also, if i'm not wrong we can get away with listening on two sockets, one of client and one for server if we go by the above approach

Reply:
The reason here is just keep original behavior in single tor, currently interface-id in single tor is not mandatory, it depends on dhcpv6 relay configuration. But in this PR, the code logic to handle responses from server support both cases. 

Do you mean we always use loopback socket for server in both dual-tor and single tor case ?

Reply:
Yeah, I meant using Loopback0 socket in both the cases.

Reply:
In current dhcpv4 behavior, single tor case use vlan ip in "giaddr" field, for dual-tor we use loopback0 in "giaddr" field. So single tor case DHCPv4 server will send response to Vlan interface, dual tor case DHCPv4 server will send response to loopback0. In this PR I want to keep consistent with DHCPv4's behavior. Since this behavior did not change original single tor behavior, it may introduce less trouble for current test cases, and safe for regression.

Reply:
I see, thanks.